File: aarrepo/src/main/java/com/zaze/aarrepo/commons/base/adapter/BaseItemHolder.java
package com.zaze.aarrepo.commons.base.adapter;

import android.view.View;

import com.marshalchen.ultimaterecyclerview.UltimateRecyclerviewViewHolder;

/**
 * Description :
 *
 * @author : ZAZE
 * @version : 2017-03-10 - 11:04
 */
public abstract class BaseItemHolder extends UltimateRecyclerviewViewHolder {

    public BaseItemHolder(View itemView, boolean isItem) {
        super(itemView);
        if (isItem) {
            initView(itemView);
        }
    }

    public BaseItemHolder(View itemView) {
        this(itemView, true);
    }

    protected abstract void initView(View itemView);
}

File: zazecommon/src/main/java/com/zaze/common/adapter/BaseItemHolder.kt
package com.zaze.aarrepo.commons.base.adapter

import android.view.View

/**
 * Description :

 * @author : ZAZE
 * *
 * @version : 2017-03-10 - 11:04
 */
abstract class BaseItemHolder(itemView: View, isItem: Boolean) : UltimateRecyclerviewViewHolder<View>(itemView) {

    init {
        if (isItem) {
            initView(itemView)
        }
    }

    protected abstract fun initView(itemView: View)
}
